flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -428,6 +455,21 @@ public enum CompressionCodec {
                                     + "tasks have occupied all the buffers and the downstream tasks are waiting for the exclusive buffers. The timeout breaks"
                                     + "the tie by failing the request of exclusive buffers and ask users to increase the number of total buffers.");
 
+    /** The timeout for requesting buffers for each channel. */
+    @Documentation.ExcludeFromDocumentation(
+            "This option is purely implementation related, and may be removed as the implementation changes.")
+    public static final ConfigOption<Duration> NETWORK_BUFFERS_REQUEST_TIMEOUT =
```

Comment:
The old `ConfigOption` can be removed as it is already covered by `withDeprecatedKeys` of the new option.

Reply:
Removing the old `ConfigOption` right away will cause the code that directly uses the `ConfigOption` to fail. It is better to mark it deprecated in 1.20 and remove it in 2.0.

Reply:
Ok, make sense.